src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -442,7 +437,9 @@ private TryUpdateSourceResults TryUpdateSource()
                 return TryUpdateSourceResults.SourceConflicted;
             }
 
-            _packageSources[_packageSources.Position] = newPackageSource;
+            selectedPackageSource.Name = name;
+            selectedPackageSource.Source = source;
+            _packageSources.ResetCurrentItem();
```

Comment:
Yes, we used to create a new object and only pass in the source/name and we would lose all the other information on the package source so when we would save it would wipe out other options like credentials, this refresh is needed to update the UI after we change the values on the existing object.